### PR TITLE
Use the SHA-256 implementation built into JavaScript engines

### PR DESCRIPTION
**Pull request by steveluscher**
This is a breaking change which _removes_ the need for browsers to introduce a polyfill for `createHash`. Engines now have a built-in `SubtleCrypto#digest` which serves this purpose, polyfill-free, with [good support](https://caniuse.com/mdn-api_subtlecrypto_digest).

The reason this is a breaking change is that `createHash` is synchronous and `SubtleCrypto#digest` is async.

Related to #7351.

### CODE REVIEW

File: libraries/type-length-value/js/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "@solana/spl-type-length-value",
     "description": "SPL Type Length Value Library",
-    "version": "0.1.0",
+    "version": "0.2.0",
```
Comment (steveluscher) [inline libraries/type-length-value/js/package.json:4]: Breaking change.

File: libraries/type-length-value/js/src/splDiscriminate.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+import { assertDigestCapabilityIsAvailable } from '@solana/assertions';
 
-export const splDiscriminate = (discriminator: string, length = 8): Buffer => {
-    const digest = createHash('sha256').update(discriminator).digest();
-    return digest.subarray(0, length);
-};
+export async function splDiscriminate(discriminator: string, length = 8): Promise<Uint8Array> {
+    assertDigestCapabilityIsAvailable();
+    const bytes = new TextEncoder().encode(discriminator);
+    const digest = await crypto.subtle.digest('SHA-256', bytes);
```
[the remark this answers is not in the document]
Comment (joncinque) [reply libraries/type-length-value/js/src/splDiscriminate.ts:6]: I think we can keep it simple and elevate the requirement for this package to Node 19. Considering it'll become a dev dependency for the two packages that use it, I can't see any drawbacks. Are there any other potential negative consequences that I'm missing?

[the remark this answers is not in the document]
Comment (steveluscher) [reply libraries/type-length-value/js/src/splDiscriminate.ts:6]: Totally.

**Review by joncinque — APPROVED** [review]
Looks great, thanks!

**Pull request merged**